docs/pai-management/doc/how-to-add-new-conm.md

```diff
@@ -0,0 +1,200 @@
+
+```bash
+src/${service_name}/config/${service_name}.py
+```
+###### Specification
+
+You should implement the parse class like following.
+
+```python
+class cluster:
+
+    def __init__(self, cluster_configuration, service_configuration, default_service_configuraiton):
+        None
+    
+    #### Fist check, ensure all the configured data in cluster_configuration, service_configuration, default_service_configuration is right. And nothing is miss.
+    def validation_pre(self):
+        #   if error: 
+        #      return False, "message"
+
+        return True, None
+    
+    #### Generate the final service object model
+    def run(self):
+        # parse your service object model here, and return a generated dictionary
+        return cluster_com
+    
+    #### All service and main module (kubrenetes, machine) is generated. And in this check steps, you could refer to the service object model which you will used in your own service, and check its existence and correctness. 
+    def validation_post(self, cluster_object_model):
+        #   if error: 
+        #      return False, "message"l
+        return True, None
+```
+
+###### The generated service object model
+
+Present python dict with yaml format
+
+```yaml
+
+service-generated-key1: generated-value1
+
+service-mandatory-key1: example-value1
+
+service-a-key1: default—value1
+
+service-a-key2: default-value2
+
+service-a-key3:
+    service-a-key3-subkey1: default-value3-subvalue1
+    service-a-key3-subkey2: default-value3-subvalue2
+    service-a-key3-subkey3: default-value3-subvalue3
+    
+service-a-key4:
+    - default-value4-sub1
+    - default-value4-sub2
+
+```
+
+#### Document <a name="#Document_link"></a>
+
+Please write a document to guide others to use the generated data of your service. The document should containers following information.
+
+- Default configuration
+
+Introduce your default configuration here.
+
+- How to configure cluster section in service-configuraiton.yaml
+
+Guide user to configure the mandatory properties and reconfigure the default.
+
+- Generated Configuration
+
+Present your generated data in yaml style
+
+- Table
+
+A table contain following content. Here I will take cluster section as an example.
+  
+<table>
+<tr>
+    <td>Data in Configuration File</td>
+    <td>Data in Cluster Object Model</td>
+    <td>Data in Jinja2 Template</td>
+    <td>Data type</td>
+</tr>
+<tr>
+    <td>cluster.common.cluster-id</td>
+    <td>com["cluster"]["common"]["cluster-id"]</td>
+    <td>com["cluster"]["common"]["cluster-id"]</td>
+    <td>string</td>
+</tr>
+</table>
+
+#### How to write your template with cluster object model <a name="#use_link"></a>
+
+
+- You could find all service configuration in corresponding services' document in the path ```src/${service_name}/config/${service_name}.md```.
+- Please refer to [jinja2 document](http://jinja.pocoo.org/docs/2.10/), and learn how to write the final tempalte with the  ```Data in Jinja2 Template``` 
```

Comment:
I have something else to discuss, which may be not related to this PR.

The python-jinja2 package installed is from [ubuntu 16.04 official repo](https://packages.ubuntu.com/xenial/python-jinja2) and the version is 2.8, doc link may should be pointed to version 2.8. Unfortunately, it has been removed from [jinja2 official site](http://jinja.pocoo.org/).

Moreover, the [`tojson filter`](http://jinja.pocoo.org/docs/2.10/templates/#tojson) introduced in jinja 2.9 might be useful to some other feartures. So are there some plans to update the version of jinja2?